Sources/Articles/Documentation.docc/2025/jwt-kit/jwt-kit.md

```diff
@@ -0,0 +1,198 @@
+
+It usually looks like this:
+```
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
+```
+
+If you inspect the token on [jwt.io](https://jwt.io), you'll see that it consists of three parts separated by dots: the header, the payload, and the signature.
+
+> Note: don't paste production tokens into jwt.io or any other online tool, as they could be intercepted and misused.
+
+## Installing JWTKit
+
+JWTKit is available as a Swift package, which makes it easy to install and use in your projects. To install it, add it as a dependency in your `Package.swift` file:
+
+```swift
+dependencies: [
+    .package(url: "https://github.com/vapor/jwt-kit.git", from: "5.0.0")
+]
+```
+
+Then, add JWTKit as a dependency to your target:
+
+```swift
+targets: [
+    .target(name: "YourTarget", dependencies: [
+        .product(name: "JWTKit", package: "jwt-kit")
+    ])
+]
+```
+
+## Configuration
+
+In JWTKit everything revolves around the ``JWTKeyCollection`` object. As the name suggests, it's a collection of keys that can be used to sign and verify JWTs. The declaration is simple, you can add this in your app's configuration code:
+
+<!-- @Snippet(path: "site/Snippets/JWTKit", slice: key_collection_init) -->
+
+```swift
+let keyCollection = JWTKeyCollection()
+```
+
+If you know the signing algorithm you want to use, adding keys to the collection is also straightforward:
+
+<!-- @Snippet(path: "site/Snippets/JWTKit", slice: key_collection_add_hmac) -->
+
+```swift
+await keyCollection.add(hmac: "secret", digestAlgorithm: .sha256)
+```
+
+This snippet adds an HMAC (Hash-based Message Authentication Code) key to the collection. HMAC works like a digital signature for our JWT, but with a simpler approach than asymmetric algorithms:
+1. First, the JWT's content (header and payload) is hashed using SHA-256
+2. Then, this hash is combined with our secret key to create a unique signature
+3. The same secret key must be used for both signing and verification (that's what makes it "symmetric")
+Think of it like a special seal on a letter - only someone with the same seal (our secret key) can verify if the letter is authentic. In this case, SHA-256 is our digest algorithm - it's responsible for creating that initial hash of our data. It's widely used because it provides a good balance of security and performance.
+
+> Important: Since HMAC uses the same key for signing and verification, you must keep this secret key secure and never expose it in client-side code or public repositories. Use environment variables or a secure secrets manager to store it.
+
+> You might be wondering why the `add` method is marked with `await`. This is because the method is asynchronous due to `JWTKeyCollection` being an actor. You can read more about actors in [Joannis' article](https://swiftonserver.com/structured-concurrency-and-shared-state-in-swift/#Actors%20and%20Isolation).
```

Comment:
```suggestion
> You might be wondering why the `add` method is marked with `await`. This is because the method is asynchronous due to `JWTKeyCollection` being an actor. You can read more about actors in <doc:structured-concurrency-and-shared-state-in-swift>
```